dockerclient.go

```diff
@@ -285,13 +286,26 @@ func (client *DockerClient) PullImage(name string, auth *AuthConfig) error {
 	v := url.Values{}
 	v.Set("fromImage", name)
 	uri := fmt.Sprintf("/%s/images/create?%s", APIVersion, v.Encode())
-
-	headers := make(map[string]string)
+	req, err := http.NewRequest("POST", client.URL.String()+uri, nil)
 	if auth != nil {
-		headers["X-Registry-Auth"] = auth.encode()
+		req.Header.Add("X-Registry-Auth", auth.encode())
 	}
-	_, err := client.doRequest("POST", uri, nil, headers)
-	return err
+	resp, err := client.HTTPClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	scanner := bufio.NewScanner(resp.Body)
+	for scanner.Scan() {
```

Comment:
you're right, the last line will have the error if there is one: https://github.com/docker/docker/blob/487a417d9fd074d0e78876072c7d1ebfd398ea7a/api/server/server.go#L518
I'll go ahead and make this change